src/main/java/service/impl/ShopServiceImpl.java

```
@@ -0,0 +1,32 @@
+package service.impl;
+
+import java.util.List;
+import model.FruitTransaction;
+import service.ShopService;
+import strategy.OperationHandler;
+import strategy.OperationStrategy;
+
+public class ShopServiceImpl implements ShopService {
+
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
```

This violates the requirement to handle invalid inputs (see examples listing: "Null parameters"). The constructor `ShopServiceImpl(OperationStrategy operationStrategy)` does not validate that `operationStrategy` is non-null. Add a null-check and throw an informative IllegalArgumentException if null so the service fails fast and clearly.